lib/new_relic/agent/instrumentation/notifications_subscriber.rb

```diff
@@ -47,6 +47,30 @@ def self.subscribe(pattern)
           end
         end
 
+        def start(name, id, payload)
+          return unless state.is_execution_traced?
+
+          start_segment(name, id, payload)
+        rescue => e
+          log_notification_error(e, name, 'start')
+        end
+
+        def finish(name, id, payload)
+          return unless state.is_execution_traced?
+
+          finish_segment(id, payload)
+        rescue => e
+          log_notification_error(e, name, 'finish')
+        end
+
+        def start_segment(name, id, payload)
+          raise NotImplementedError
+        end
+
+        def finish_segment(name, id, payload)
```

Comment:
Looks like `finish_segment` doesn't ever call `name`, so it could remove as an arg!